basic/bittwister_test.go

```diff
@@ -0,0 +1,609 @@
+	require.NoError(t, executor.WaitInstanceIsRunning(), "Error waiting for executor instance to be running")
+
+	// Perform the test
+
+	tt := []struct {
+		name                 string
+		targetPacketlossRate int32
+		tolerancePercent     int
+	}{{
+		name:                 "10%",
+		targetPacketlossRate: 10,
+		tolerancePercent:     50,
+	},
+		{
+			name:                 "20%",
+			targetPacketlossRate: 20,
+			tolerancePercent:     30,
+		},
+		{
+			name:                 "30%",
+			targetPacketlossRate: 30,
+			tolerancePercent:     10,
+		},
+		{
+			name:                 "50%",
+			targetPacketlossRate: 50,
+			tolerancePercent:     10,
+		},
+		{
+			name:                 "70%",
+			targetPacketlossRate: 70,
+			tolerancePercent:     10,
+		},
+		{
+			name:                 "90%",
+			targetPacketlossRate: 90,
+			tolerancePercent:     10,
+		},
+		{
+			name:                 "100%",
+			targetPacketlossRate: 100,
+			tolerancePercent:     10,
+		},
+	}
+
+	targetIP, err := target.GetIP()
+	require.NoError(t, err, "Error getting IP")
+
+	for _, tc := range tt {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			err = target.SetPacketLoss(tc.targetPacketlossRate)
+			require.NoError(t, err, "Error setting packetloss rate")
+
+			t.Log("Starting packetloss test. It takes a while.")
+			startTime := time.Now()
+			ctx, cancel := context.WithTimeout(context.Background(), commandTimeout)
+			defer cancel()
+
+			targetAddress := fmt.Sprintf("%s:%d", targetIP, gopingPort)
+			output, err := executor.ExecuteCommandWithContext(ctx, "goping", "ping", "-q",
+				"-c", fmt.Sprint(numOfPingPackets),
+				"-t", packetTimeout.String(),
+				"-m", "packetloss",
+				targetAddress)
+			require.NoError(t, err, "Error executing command")
+
+			elapsed := time.Since(startTime)
+			t.Logf("Test took %d seconds", int64(elapsed.Seconds()))
+
+			gotPacketloss, err := strconv.ParseFloat(output, 64)
+			require.NoError(t, err, "Error parsing output")
+
+			deviationPercent := math.Abs(gotPacketloss-float64(tc.targetPacketlossRate)) / float64(tc.targetPacketlossRate) * 100
+			assert.LessOrEqual(t, deviationPercent, float64(tc.tolerancePercent), "Deviation is too high")
+
+			t.Logf("Packetloss expected: %v%% \tgot: %.2f%% \tdeviation: %.2f%% \ttolerance: %v%%",
+				tc.targetPacketlossRate,
+				gotPacketloss,
+				deviationPercent,
+				tc.tolerancePercent)
+		})
+	}
+}
+func TestBittwister_Latency(t *testing.T) {
+	t.Parallel()
+	// Setup
+
+	const (
+		numOfPingPackets = 100
+		packetTimeout    = 100 * time.Millisecond
+		commandTimeout   = 10 * time.Minute
+	)
+
+	mother, err := knuu.NewInstance("mother")
+	require.NoError(t, err, "Error creating instance")
+
+	require.NoError(t, mother.EnableBitTwister(), "Error enabling BitTwister")
+	err = mother.SetImage(gopingImage)
+	require.NoError(t, err, "Error setting image")
+
+	gopingPort := 8001
+	err = mother.SetEnvironmentVariable("SERVE_ADDR", fmt.Sprintf("0.0.0.0:%d", gopingPort))
```

Comment:
optimization: If you set the env variable after you commit, the Instance is prepared faster, as it does not need to build a container image.

Reply:
done